editor/src/clj/editor/outline_view.clj

```diff
@@ -458,9 +463,29 @@
       (ui/cancel future)
       (ui/user-data! cell :future-expand nil))))
 
+(defn- toggle-visibility! [node-outline-key-path ^MouseEvent event]
+  (ui/run-command (.getSource event) :hide-toggle {:node-outline-key-path node-outline-key-path})
+  (.consume event))
+
+(def eye-open-path (ui/load-svg-path "scene/images/eye_open.svg"))
+(def eye-closed-path (ui/load-svg-path "scene/images/eye_closed.svg"))
+
 (defn make-tree-cell
   [^TreeView tree-view drag-entered-handler drag-exited-handler]
-  (let [cell (proxy [TreeCell] []
+  (let [eye-icon-open (app-view/make-svg-icon-graphic eye-open-path)
+        eye-icon-closed (app-view/make-svg-icon-graphic eye-closed-path)
```

Comment:
Part of the initialization was moved here for performance purposes. That solved almost all blinking issues on visibility toggling, but there are still some rare cases where a tree cell might appear to be re-rendered, although it shouldn't. Is there anything else we can do?

Reply:
I suspect there might be a bug in the JavaFX redraw code. I've certainly seen issues around hover highlighting and the disclosure triangle earlier. I don't think we should worry about it for this PR.